core/file_walker.go

```diff
@@ -40,17 +37,16 @@ func (f *File) UpdateSize() {
 
 type ReadDir func(dirname string) ([]os.FileInfo, error)
 
-func GetSubTree(path string, parent *File, readDir ReadDir, ignoredFolders map[string]struct{}) *File {
-	var mutex sync.Mutex
+func WalkFolder(path string, readDir ReadDir, ignoredFolders map[string]struct{}) *File {
```

Comment:
This is no longer recursive method, might as well have non-recursive name and interface